lifecycle-operator/apis/lifecycle/v1beta1/keptntask_types.go

```diff
@@ -0,0 +1,278 @@
+	// considered to be failed.
+	// +optional
+	// +kubebuilder:default:="5m"
+	// +kubebuilder:validation:Pattern="^0|([0-9]+(\\.[0-9]+)?(ns|us|µs|ms|s|m|h))+$"
+	// +kubebuilder:validation:Type:=string
+	// +optional
+	Timeout metav1.Duration `json:"timeout,omitempty"`
+}
+
+type TaskContext struct {
+	// WorkloadName the name of the KeptnWorkload the KeptnTask is being executed for.
+	// +optional
+	WorkloadName string `json:"workloadName"`
+	// AppName the name of the KeptnApp the KeptnTask is being executed for.
+	// +optional
+	AppName string `json:"appName"`
+	// AppVersion the version of the KeptnApp the KeptnTask is being executed for.
+	// +optional
+	AppVersion string `json:"appVersion"`
+	// WorkloadVersion the version of the KeptnWorkload the KeptnTask is being executed for.
+	// +optional
+	WorkloadVersion string `json:"workloadVersion"`
+	// TaskType indicates whether the KeptnTask is part of the pre- or postDeployment phase.
+	// +optional
+	TaskType string `json:"taskType"`
+	// ObjectType indicates whether the KeptnTask is being executed for a KeptnApp or KeptnWorkload.
+	// +optional
+	ObjectType string `json:"objectType"`
+}
+
+type TaskParameters struct {
+	// Inline contains the parameters that will be made available to the job
+	// executing the KeptnTask via the 'DATA' environment variable.
+	// The 'DATA'  environment variable's content will be a json
+	// encoded string containing all properties of the map provided.
+	// +optional
+	Inline map[string]string `json:"map,omitempty"`
+}
+
+type SecureParameters struct {
+	// Secret contains the parameters that will be made available to the job
+	// executing the KeptnTask via the 'SECRET_DATA' environment variable.
+	// The 'SECRET_DATA'  environment variable's content will the same as value of the 'SECRET_DATA'
+	// key of the referenced secret.
+	// +optional
+	Secret string `json:"secret,omitempty"`
+}
+
+// KeptnTaskStatus defines the observed state of KeptnTask
+type KeptnTaskStatus struct {
+	// JobName is the name of the Job executing the Task.
+	// +optional
+	JobName string `json:"jobName,omitempty"`
+	// Status represents the overall state of the KeptnTask.
+	// +kubebuilder:default:=Pending
+	// +optional
+	Status common.KeptnState `json:"status,omitempty"`
+	// Message contains information about unexpected errors encountered during the execution of the KeptnTask.
+	// +optional
+	Message string `json:"message,omitempty"`
+	// StartTime represents the time at which the KeptnTask started.
+	// +optional
+	StartTime metav1.Time `json:"startTime,omitempty"`
+	// EndTime represents the time at which the KeptnTask finished.
+	// +optional
+	EndTime metav1.Time `json:"endTime,omitempty"`
+	// Reason contains more information about the reason for the last transition of the Job executing the KeptnTask.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:storageversion
```

Comment:
please move the `storageversion` to the task in v1alpha3 - the storage versions will change in a follow up ticket when we adapt the controller to use v1beta1